### PR TITLE
Retiring `text_bytes_aware_shuffle` to use `shuffle` directly

### PR DESCRIPTION
**Pull request by praateekmahajan**
## Description
Now that cudf supports large strings, we can use shuffle directly. 
Resolves https://github.com/NVIDIA/NeMo-Curator/issues/240 

Here is the performance improvement when compared existing older implementation (which ultimately called `rearrange_column_by_direct`)
image

## Usage

```python
# Add snippet demonstrating usage
```
## Checklist

- [ ] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [ ] New or Existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by sarahyurick** [comment]
Thanks! Was wondering if you have any small-scale examples that could be added to `test_fuzzy_dedup.py`?

**Review by VibhuJawa — COMMENTED** [review]
This looks good to me on the assumption we have verified `output_shuffled_docs_path` looked the same for both `main` and  the `PR` .

**Review by VibhuJawa — APPROVED** [review]

File: nemo_curator/modules/fuzzy_dedup.py [hunk 1]
```diff
@@ -1140,36 +1142,65 @@ def _batched_merge_and_write(
...
-                    if parts_per_text_batch_retry < 1:
-                        raise err
-                    print(
-                        f"\nWe encountered an OverflowError and will retry "
-                        f"the current batch with {parts_per_text_batch_retry} "
-                        f"text partitions instead of {parts_per_text_batch_use}.",
-                        flush=True,
+                elif os.environ["SHUFFLE_APPROACH"] == "dask_vanilla":
+                    self._logger.info("Using dask's vanilla shuffle")
+                    output_df = subset_merged_df.shuffle(on=partition_on)
```
[the remark this answers is not in the document]
Comment (praateekmahajan) [reply nemo_curator/modules/fuzzy_dedup.py:1160]: @sarahyurick we won't be keeping all approaches, I just had them for my own benchmarking efforts. I'll update the PR with the results, and based on those we'll only be keeping the approach of `dask_vanilla` (once @ayushdg is able to confirm at scale too)

**Review by sarahyurick — APPROVED** [review]
LGTM!

**Pull request merged**